distributions/test/src/test/java/com/connexta/ion/replication/distributions/test/IntegrationTest.java

```diff
@@ -0,0 +1,149 @@
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.Network;
+
+// TODO: add javadoc
+public abstract class IntegrationTest {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(IntegrationTest.class);
+
+  private static final String RESOURCE_VARIABLE_DELIMETER = "$";
+
+  public static final String CATALOG_ENDPOINT = "/catalog";
+
+  public static final String CONTENT_TYPE = "Content-Type";
+
+  public static final String APPLICATION_XML = "application/xml";
+
+  public static final String APPLICATION_JSON = "application/json";
+
+  public static final String CONFIG = "config";
+
+  public static final String SITE = "site";
+
+  public static final String STATUS = "status";
+
+  public static final String ITEM = "item";
+
+  private static final int SOLR_PORT = 8983;
+
+  @ClassRule public static Network network = Network.newNetwork();
+
+  @ClassRule
+  public static GenericContainer solr =
+      new GenericContainer("r.ion.phx.connexta.com/replication-solr:0.3.0-SNAPSHOT")
+          .withNetwork(network)
+          .withNetworkAliases("replication-solr")
+          .withExposedPorts(SOLR_PORT);
+
+  @ClassRule
+  public static GenericContainer replication =
+      new GenericContainer("r.ion.phx.connexta.com/replication:0.3.0-SNAPSHOT")
+          .withNetwork(network)
+          .withNetworkAliases("ion-replication")
+          .withClasspathResourceMapping(
+              "serverKeystore.jks",
+              "/opt/replication/config/keystores/serverKeystore.jks",
+              BindMode.READ_ONLY)
+          .withClasspathResourceMapping(
+              "serverTruststore.jks",
+              "/opt/replication/config/keystores/serverTruststore.jks",
+              BindMode.READ_ONLY)
+          .withClasspathResourceMapping(
+              "ssl.properties", "/opt/replication/config/ssl.properties", BindMode.READ_ONLY)
+          .withClasspathResourceMapping(
+              "application.yml", "/opt/replication/config/application.yml", BindMode.READ_ONLY);
+
+  public static String solrUrl;
+
+  public static String sourceUrl;
+
+  public static String destinationUrl;
+
+  public static InputStream getFileAsStream(String filePath) {
+    return IntegrationTest.class.getClassLoader().getResourceAsStream(filePath);
+  }
+
+  /**
+   * Retrieves a file and returns the contents within as a string. The filePath is assumed to be
+   * relative to the {@link IntegrationTest} class.
+   *
+   * @param filePath the path of the file relative to the {@link IntegrationTest} class
+   * @return A String containing the contents of the file
+   */
+  public static String getFileAsText(String filePath) {
+    return getFileAsText(filePath, ImmutableMap.of());
+  }
+
+  /**
+   * Retrieves a file and returns the contents within as a string. The filePath is considered
+   * relative to the class provided. Text in the file can be marked for variable replacement by
```

Comment:
✏️ Javadoc says relative to the class provided, but the IntegrationTest class is used.